src/dmd/link.d

```diff
@@ -151,6 +172,37 @@ version (Posix)
     }
 }
 
+/**
+Use to restore an environment variable when this structure goes out of scope.
+*/
+struct EnvRestore(string varName)
+{
+    private string _saved;
+    private bool restore;
+
+    /**
+    Access the saved environment variable value.
+    */
+    string saved() const { return _saved; }
+
+    /**
+    Call before the overriding the environment variable to save and restore it later.
+    */
+    void saveToRestoreLater()
```

Comment:
How about wrapping `EnvRestore` as a voldemort type in a function and thus doing this in at construction?

Reply:
it doesn't always save the environment variable, only in some cases.